CHANGELOG.md

```diff
@@ -243,17 +243,23 @@ Additions to existing modules
   ```agda
   _≰_ : Rel (Fin n) 0ℓ
   _≮_ : Rel (Fin n) 0ℓ
+  lower : ∀ (i : Fin m) → .(toℕ i ℕ.< n) → Fin n
   ```
 
 * In `Data.Fin.Permutation`:
   ```agda
   cast-id : .(m ≡ n) → Permutation m n
-  swap : Permutation m n → Permutation (suc (suc m)) (suc (suc n))
+  inject!-injective : swap : Permutation m n → Permutation (suc (suc m)) (suc (suc n))
   ```
 
 * In `Data.Fin.Properties`:
   ```agda
-  cast-involutive : .(eq₁ : m ≡ n) .(eq₂ : n ≡ m) → ∀ k → cast eq₁ (cast eq₂ k) ≡ k
+  cast-involutive                  : .(eq₁ : m ≡ n) .(eq₂ : n ≡ m) → ∀ k → cast eq₁ (cast eq₂ k) ≡ k
+  inject!-injective                : Injective _≡_ _≡_ inject!
+  inject!-<                        : (k : Fin′ i) → inject! k < i
+  lower-injective                  : lower i i<n ≡ lower j j<n → i ≡ j
+  injective⇒nonStrictlyContractive : ∀ (f : Fin n → Fin m) → Injective _≡_ _≡_ f → ∀ i → ¬ (∀ j → j ≤ i → f j < i)
```

Comment:
Suggest leaving this out of the PR: as below, it's an easy corollary of `injective⇒existsPivot` after re-ordering.